tests/integrationv2/test_external_psk.py

```diff
@@ -0,0 +1,294 @@
+import copy
+import os
+import pytest
+import time
+import enum
+
+from configuration import available_ports
+from common import ProviderOptions, Protocols, data_bytes, Ciphers, Certificates
+from fixtures import managed_process
+from providers import Provider, S2N, OpenSSL
+from utils import invalid_test_parameters, get_parameter_name, to_bytes
+from collections import namedtuple
+from enum import Enum, auto
+
+# Test Vectors from https://tools.ietf.org/html/rfc8448#section-4
+shared_psk_identity = '2c035d829359ee5ff7af4ec900000000262a6494dc486d2c8a34cb33fa90bf1b00'\
+                      '70ad3c498883c9367c09a2be785abc55cd226097a3a982117283f82a03a143efd3'\
+                      'ff5dd36d64e861be7fd61d2827db279cce145077d454a3664d4e6da4d29ee03725'\
+                      'a6a4dafcd0fc67d2aea70529513e3da2677fa5906c5b3f7d8f92f228bda40dda72'\
+                      '1470f9fbf297b5aea617646fac5c03272e970727c621a79141ef5f7de6505e5bfb'\
+                      'c388e93343694093934ae4d357'
+
+shared_psk_secret = '4ecd0eb6ec3b4d87f5d6028f922ca4c5851a277fd41311c9e62d2c9492e1c4f3'
+
+s2n_known_value_psk_sha256 = shared_psk_identity + ',' + shared_psk_secret + ',' + 'SHA256'
+
+s2n_known_value_psk_sha384 = shared_psk_identity + ',' + shared_psk_secret + ',' + 'SHA384'
+
+openssl_known_value_psk_parameters = [ '-psk_identity', shared_psk_identity, '--psk', shared_psk_secret ]
+
+
+# Arbitrary Test vectors
+s2n_psk_parameters_sha384 = 'psk_identity_sha384,'\
+                            'a6dadae4567876,'\
+                            'SHA384'\
+
+s2n_psk_parameters_sha256 = 'psk_identity_sha256,'\
+                            'a64dafcd0fc67d2a,'\
+                            'SHA256'\
+
+
+# PSK Identity marker to assert with openssl logs
+psk_identity_marker_openssl_client = [ b"2c0",
+                                       b"35d829359ee5ff7a", b"f4ec900000000262", b"a6494dc486d2c8a3", 
+                                       b"4cb33fa90bf1b007", b"0ad3c498883c9367", b"c09a2be785abc55c", 
```

Comment:
These are pretty intense. What are they, and how did you calculate them?

Reply:
The OpenSSL logs contain the psk identity split across different lines - which is the split, this is not calculated and is the same psk identity sent from the peer.